src/setup/rulerPortraits.ts

```diff
@@ -0,0 +1,43 @@
+import portrait_0 from '../assets/ui/ruler_portraits/portrait_0.png';
+import portrait_1 from '../assets/ui/ruler_portraits/portrait_1.png';
+import portrait_2 from '../assets/ui/ruler_portraits/portrait_2.png';
+import portrait_3 from '../assets/ui/ruler_portraits/portrait_3.png';
+import portrait_4 from '../assets/ui/ruler_portraits/portrait_4.png';
+import portrait_5 from '../assets/ui/ruler_portraits/portrait_5.png';
+import portrait_6 from '../assets/ui/ruler_portraits/portrait_6.png';
+import portrait_7 from '../assets/ui/ruler_portraits/portrait_7.png';
+import portrait_8 from '../assets/ui/ruler_portraits/portrait_8.png';
+import portrait_9 from '../assets/ui/ruler_portraits/portrait_9.png';
+import portrait_10 from '../assets/ui/ruler_portraits/portrait_10.png';
+import portrait_11 from '../assets/ui/ruler_portraits/portrait_11.png';
+import portrait_12 from '../assets/ui/ruler_portraits/portrait_12.png';
```

Comment:
these could be in a for loop:
for (let i = 0; i <= 12; i++) {
  import(`../assets/ui/ruler_portraits/portrait_${i}.png`)

same with portraitSources below